UnityProject/Assets/Scripts/Objects/Machines/Gibber.cs

```diff
@@ -0,0 +1,163 @@
+	{
+		[SerializeField] private ObjectContainer storage;
+		[SerializeField] private UniversalObjectPhysics physics;
+		[SerializeField] private Integrity machineIntegrity;
+		[SerializeField] private float timeToGib = 12f;
+		[SerializeField] private int produceMultiplier = 4;
+		[SerializeField] private int damagePerFrame = 20;
+		[SerializeField] private GameObject defaultProduce;
+		[SerializeField] private SpriteHandler lights;
+		[SerializeField] private SpriteDataSO lightsoff;
+		[SerializeField] private SpriteDataSO lightson;
+
+		private bool isRunning = false;
+
+		private Dictionary<GameObject, int> gibbed = new Dictionary<GameObject, int>();
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			// Only interact with this if it's anchored.
+			if (physics.isNotPushable == false) return false;
+			//Checks if the player is in reach + if they are able to interact with this type of object.
+			if (DefaultWillInteract.Default(interaction, side) == false) return false;
+			//For people attacking the object so they don't accidentally trip it.
+			return interaction.Intent != Intent.Harm;
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (interaction.HandObject != null)
+			{
+				storage.StoreObject(interaction.HandObject);
+				return;
+			}
+
+			isRunning = !isRunning;
+			if (isRunning) StartCoroutine(GibbingTime());
+		}
+
+		public bool WillInteract(MouseDrop interaction, NetworkSide side)
+		{
+			// Only interact with this if it's anchored.
+			if (physics.isNotPushable == false) return false;
+			//Checks if the player is in reach + if they are able to interact with this type of object.
+			return DefaultWillInteract.Default(interaction, side);
+		}
+
+		public void ServerPerformInteraction(MouseDrop interaction)
+		{
+			storage.StoreObject(interaction.DroppedObject);
+		}
+
+		private IEnumerator GibbingTime()
+		{
+			Chat.AddLocalMsgToChat("The gibber violently shakes as it shreds everything inside of it.", gameObject);
+			lights.SetSpriteSO(lightson);
+			int time = 0;
+			while (isRunning)
+			{
+				if(timeToGib <= time) break;
+				//Always update the list to avoid collection NREs when moving things in and out of storage.
+				time++;
+				yield return WaitFor.Seconds(0.6f);
+				yield return CheckContentAndHarm();
+			}
+
+			storage.RetrieveObjects();
+			foreach (var products in gibbed)
+			{
+				_ = Spawn.ServerPrefab(products.Key, gameObject.TileWorldPosition().To3(), gameObject.RegisterTile().Matrix.transform,
+					null, Mathf.Min(1, products.Value));
+			}
+			gibbed.Clear();
+			isRunning = false;
+			Chat.AddLocalMsgToChat("The gibber stops vibrating as it finishes it's operation.", gameObject);
+			lights.SetSpriteSO(lightsoff);
+		}
+
+		private IEnumerator CheckContentAndHarm()
+		{
+			var list = storage.GetStoredObjects().ToArray();
```

Comment:
No need to create an array object on each tick here, we can iterate directly on the method return.

Reply:
Enumeration errors when the list changes are a thing

Reply:
You can just backwards for loop to avoid the enumeration error with lists

Reply:
Apparently `.Reverse()` works the same after testing it. no NREs